doc/contributing/adding-v8-fast-api.md

```diff
@@ -0,0 +1,119 @@
+# Adding V8 Fast API
+
+Node.js uses [V8](https://github.com/v8/v8) as the JavaScript engine.
+In order to provide fast paths for functions that are called quite often,
+V8 proposes the usage of `fast api calls` that does not use any of the V8
+internals, but uses internal C functions.
+
+## Limitations
+
+- Fast api calls can not use `V8` internals inside the fast path.
+- Not all parameter and return types are supported in fast api calls.
+For a full list, please look into `v8-fast-api-calls.h` file.
+
+## Requirements
+- Each unique fast path function signature should be defined inside
+`node_external_reference.h` file.
+- In order to test fast paths, make sure to run the tests in a loop
+with more than 1000 iterations to force V8 to optimize and prefer the
+fast api over slow path.
+
+## Fallback to slow path
+
+Fast apis support fallback to slow path (implementation that uses V8 internals)
+in case logically it is wise to do so. Fallback mechanism can be enabled, and changed
```

Comment:
```suggestion
in case logically it is wise to do so, for example when providing a more detailed error. Fallback mechanism can be enabled, and changed
```